src/treelearner/serial_tree_learner.cpp

```diff
@@ -172,18 +173,29 @@ Tree* SerialTreeLearner::Train(const score_t* gradients, const score_t *hessians
   int cur_depth = 1;
   // only root leaf can be splitted on first time
   int right_leaf = -1;
-  for (int split = 0; split < tree_config_->num_leaves - 1; ++split) {
+
+  int init_splits = 0;
+  bool aborted = false;
+  if (forced_split_json != nullptr) {
+    init_splits = ForceSplits(tree.get(), forced_split_json, &left_leaf,
```

Comment:
it seems the force_split is only used in the root node, is that correct?

Reply:
I find there is loop inside `ForceSplits`, however, the num_leaves is only reduced by 1 here.